lib/internal/main/check_syntax.js

```diff
@@ -75,5 +75,5 @@ async function checkSyntax(source, filename) {
     return;
   }
 
-  wrapSafe(filename, source, undefined, 'commonjs');
+  wrapSafe(filename, source);
```

Comment:
I think you can just remove the module format detection stuff above if you are already using wrapSafe to handle both formats.

Reply:
What do you mean exactly? There’s no syntax detection stuff above this. The other module format determination logic predated `detect-module`.

Reply:
Basically, don't do `isModule` and just use a `format`, which can be undefined; It's being broken because you are changing the return value of `defaultGetFormat()` to be possibly undefined, then just pass that to wrapSafe.

Reply:
So like this? [`7e0424a` (#53619)](https://github.com/nodejs/node/pull/53619/commits/7e0424a05b4f1b55a33305790b6b1a83494aede3)